submissions/SofiiaTrokhymchuk/tiny-js-oop/index.js

```diff
@@ -0,0 +1,88 @@
+class Inhabitant{
+   constructor(species, name, gender, saying, legs){
+      this.species = species;
+      this.name = name;
+      this.gender = gender;
+      this.saying = saying;
+      this.legs = legs;
+      this.friends = [];
+   }
+
+   getProperties(){
+      return ['species', 'name', 'gender', 'saying', 'legs', 'friends'];
+   }
+
+   addFriends(friendsList){
+      this.friends = this.friends.concat(friendsList);
+   }
+
+   friendsToString(){
+      return this.friends.length === 0 ? '0' : this.friends.map(friend => friend.name).join(', ');
+   }
+
+   toString(){
+      return this.getProperties().map(property => {
+         return `<strong>${property}</strong>: ${property === 'friends' ? this.friendsToString() : this[property]}`;
```

Comment:
Not really scalable approach... Imagine, you have more array properties (parents, children, pets).
Current approach will make this code really unreadable.
Better approach would be convert all arrays (currently one) into strings and then just process strings.
This would require to change the entire setup of your classes.
In child classes you could overload not `getProperties` but `toString` where you'd call `super.toString()` and add whatever is needed to add about hands.
This would bring you to the fact that you do not need `getProperties` at all.
As a result you would have had less code, strict encapsulation of properties and behaviour.

You need to master data manipulations. You will need this.

Reply:
Thank You so much for the feedback. I thought that making the` getProperties()` function will make my code more scalable and I could use an array of properties somewhere else not only inside `toString()` function. Now I understand that I confused myself instead.

Reply:
> I thought that making the` getProperties()` function will make my code more scalable and I could use an array of properties somewhere else not only inside `toString()` function.

This is a valid **assumption**. You will solve this problem when **it arrives** (although not necessary with current approach). Otherwise it is a so called "premature optimization". 

Prefer principles when the choice is between premature optimization or following OOP principles (encapsulation in this case) or any other fundamental software development principles and patterns.

Trade them only when optimization is required and cannot be done without violating the principles.

Reply:
It is good that you think about optimizations and re-usability.
You will (and already are) doing well.